### PR TITLE
Integrate PostCommitValidation with JdbcLedgerDao and SqlLedger

### LINKED ISSUE #5035: Drop old participant schema
Right now we are storing the transactions in the new ready-optimized schema along with the old full "ledger". Once the `SqlLedger` can be dropped, indexing should be moved directly into the indexer, without having to be called via the `JdbcLedgerDao`.

_Originally posted by @SamirTalwar-DA in https://github.com/digital-asset/daml/pull/5029/files_

### LINKED ISSUE #5663: Streamline post-commit validation
Post-commit validation is performed to ensure that parallel execution of commands does not cause inconsistent results, which include:
- double spends
- duplicate keys
- violations of causal monotonicity

These errors are laid out as `SequencingError`s.

The new sandbox actually performs this checks twice, once through the DAML-on-X architecture and once again via the `ActiveLedgerStateManager` which is used to perform the same checks in the now deprecated implementation of the sandbox.

Refactor the checks performed by the `ActiveLedgerStateManager` so that the new sandbox can skip the second, unnecessary check.

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Closes #5035
Closes #5663

changelog_begin
[Sandbox] Skip unnecessary double post-commit validation inherited by sandbox-classic, expect performance improvement
changelog_end

### TODO

- [ ] ~Probably: parameterize description for `RejectionReasons.Inconsistent`~
- [x] [Make tests compile](https://github.com/digital-asset/daml/pull/5781/commits/a1339766cd6aa1c59ae35eabc47d962356dc4c15)
- [x] [Pass tests](https://github.com/digital-asset/daml/pull/5781/commits/ff6321fc6acd63ceeb9c99ed4fced18d643fe7f5)
- [x] [Data migration to drop checkpoints](https://github.com/digital-asset/daml/pull/5781/commits/8072f8fcb1c79b49368125b0826d74a532f25690)
- [x] [Schema migration to drop old schema columns](https://github.com/digital-asset/daml/pull/5781/commits/8072f8fcb1c79b49368125b0826d74a532f25690)

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/MeteredLedgerDao.scala [hunk 1]
```diff
@@ -163,25 +152,54 @@ class MeteredLedgerDao(ledgerDao: LedgerDao, metrics: MetricRegistry)
     val storePartyEntry: Timer = metrics.timer(prefix :+ "store_party_entry")
     val storeInitialState: Timer = metrics.timer(prefix :+ "store_initial_state")
     val storePackageEntry: Timer = metrics.timer(prefix :+ "store_package_entry")
-    val storeLedgerEntry: Timer = metrics.timer(prefix :+ "store_ledger_entry")
+    val storeTransaction: Timer = metrics.timer(prefix :+ "store_ledger_entry")
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/MeteredLedgerDao.scala:155]: Wanna rename this one?

```suggestion
    val storeTransaction: Timer = metrics.timer(prefix :+ "store_transaction")
```

Comment (gerolf-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/MeteredLedgerDao.scala:155]: This is to retain the current metric name, to not cause disturbance in metric dashboards.

Comment (SamirTalwar-DA) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/MeteredLedgerDao.scala:155]: Eh, fair enough.

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/MeteredLedgerDao.scala:155]: I did it on purpose, I believe @mziolekda wants them to be stable until we have a plan to change them all at once with detailed documentation.

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/MeteredLedgerDao.scala:155]: Well, hopefully the only disturbance will be the smaller numbers from the same metric. :stuck_out_tongue:

**Comment by stefanobaghino-da** [comment]
Self-review: implicit party allocation is missing.

**Comment by stefanobaghino-da** [comment]
Weird: I just tried if implicit party allocation works using DAML Script and it does. Despite this, `ImplicitPartyAdditionIT` keeps failing.

**Comment by stefanobaghino-da** [comment]
> Weird: I just tried if implicit party allocation works using DAML Script and it does. Despite this, `ImplicitPartyAdditionIT` keeps failing.

Find out why: I've enabled implicit party allocation via the API service but `ImplicitPartyAdditionIT` doesn't use a full server but only the ledger interface.

**Comment by stefanobaghino-da** [comment]
For reviewers: currently the `Stop deduplicating commands on completion failure` conformance error is flaky. Around half of the times it causes a witness to be written for a non-existing contract, causing a foreign key constraint error. I'm investigating the issue but please do consider paying extra attention to this detail.

**Comment by stefanobaghino-da** [comment]
Thanks to input from @gerolf-da, https://github.com/digital-asset/daml/pull/5781/commits/165e261d9cacc9f432f1d5cc19a4394eaac26ce1 seems to have solved the issue. This however, introduced a relevant slowdown (2x) in conformance tests. I'll investigate the issue to see if there's something we can do about it

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala [hunk 2]
```diff
@@ -288,10 +288,8 @@ final class SandboxServer(
         ledgerConfiguration = ledgerConfiguration,
         commandConfig = config.commandConfig,
         partyConfig = PartyConfiguration.default.copy(
-          // In this version of Sandbox, parties are always allocated implicitly. Enabling
-          // this would result in an extra `writeService.allocateParty` call, which is
-          // unnecessary and bad for performance.
-          implicitPartyAllocation = false,
+          // sandbox-classic always allocates party implicitly
+          implicitPartyAllocation = true,
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala:292]: Previously `ActiveLedgerStateManager` did the implicit party allocation. But now that it's gone, we resort to the same mechanism as sadbonx (i.e. explicit implicit party allocation), right?

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala:292]: Precisely, if this flag is on the command submission service walks the transaction tree and does the party allocation.

**Pull request merged**